Generator/Sources/NeedleFramework/Generating/DependencyProviderDeclarerTask.swift

```diff
@@ -34,9 +35,29 @@ class DependencyProviderDeclarerTask: AbstractTask<[DependencyProvider]> {
     ///
     /// - returns: The list of `DependencyProvider`.
     override func execute() -> [DependencyProvider] {
+        // Do not produce a provider if the component is not a root and
+        // has no parents. In this case, the component is just an orphan
+        // scope that should be ignored.
+        if component.parents.isEmpty && !component.isRoot {
+            warning("\(component.name) is an orphan scope therefore ignored from parsing.")
+            return []
+        }
+
         return ancestorPaths(for: component)
-            .map { (path: [Component]) -> DependencyProvider in
-                return DependencyProvider(path: path, dependency: component.dependency)
+            .filter { !$0.isEmpty }
```

Comment:
nit: this filter is pretty much the `guard let` + `compactMap` combo in the next 3 lines.

Reply:
Yeah you are right. The downside is I can't remove the guard since `first` returns optional. And of course we need the `compactMap`.